src/internal/SetupResumeAcceptor.cpp

```diff
@@ -59,19 +60,26 @@ SetupResumeAcceptor::SetupResumeAcceptor(
     defaultFrameSerializer_ =
         FrameSerializer::createFrameSerializer(protocolVersion);
   }
+  eventBase_ = folly::EventBaseManager::get()->getExistingEventBase();
+  CHECK(eventBase_);
 }
 
 SetupResumeAcceptor::~SetupResumeAcceptor() {
-  for (auto& connection : connections_) {
-    connection->close(std::runtime_error("SetupResumeAcceptor closed"));
-  }
+  close().get();
```

Comment:
This will still be called even if we explicitly call `close()` before and cause a thread hop, probably fine I guess.